### PR TITLE
fix: remove steps when remove build

### PR DESCRIPTION
**Pull request by minzcmu**
remove steps of the build when removing build

### CODE REVIEW

**Review by parthasl — APPROVED** [review]
👍

**Comment by minzcmu** [comment]
this is not really efficient, when you delete a pipeline with a lot of builds and a lot of steps. Need to fire a lot of DB request to do that. convert this to a story so that we can delete steps by bulk

File: lib/build.js [hunk 1]
```diff
@@ -557,6 +557,28 @@ class BuildModel extends BaseModel {
...
+                    // Done removing steps
+                    return null;
+                }
+
+                return Promise.all(steps.map(step => step.remove()))
+                    .then(() => removeSteps());
+            }));
+
+        // Remove steps
+        return removeSteps()
```
Comment (jithine) [inline lib/build.js:578]: should do this in `finally` rather than `then`

File: lib/build.js [hunk 2]
```diff
@@ -557,6 +557,28 @@ class BuildModel extends BaseModel {
...
+        const removeSteps = (() =>
+            // Iterate through the steps and remove them
+            this.getSteps().then((steps) => {
+                if (steps.length === 0) {
+                    // Done removing steps
+                    return null;
+                }
+
+                return Promise.all(steps.map(step => step.remove()))
+                    .then(() => removeSteps());
```
Comment (jithine) [inline lib/build.js:574]: why is this being done recursively ? is `steps` dynamic ?

Comment (minzcmu) [reply lib/build.js:574]: previously we have default pagination. That's why it's recursive. Closed this anyway in favor of https://github.com/screwdriver-cd/screwdriver/issues/1614

**Pull request closed without merging**